Swift/Document.swift

```diff
@@ -91,7 +91,26 @@ public class Document : Properties {
         database = inDatabase
         _docimpl = impl
         super.init(impl)
+        
+        _docimpl.swiftDocument = self
+    }
+    
+    deinit {
+        _docimpl.swiftDocument = nil
     }
 
     let _docimpl: CBLDocument
 }
+
+
+extension CBLDocument {
+    private struct AssociatedKeys {
+        static var SwiftDocument = "SwiftDocument"
+    }
+    
+    var swiftDocument: Document? {
+        get {return objc_getAssociatedObject(self, &AssociatedKeys.SwiftDocument) as? Document}
+        set { objc_setAssociatedObject(self, &AssociatedKeys.SwiftDocument, newValue,
```

Comment:
It would be more efficient to add a private property to CBLDocument for this; we own the class so we don't need to use associatedObject (which is kind of slow.)

I'm a little worried that because the reference is weak, the Document object is likely to get dealloced very quickly unless something on the Swift side is holding onto it. But it's not a correctness problem, just (possibly) performance.